src/main/java/de/unistuttgart/iste/gits/quiz_service/persistence/mapper/QuizMapper.java

```diff
@@ -34,31 +31,103 @@ public Quiz entityToDto(QuizEntity entity) {
 
     private Question questionEntityToDto(QuestionEntity questionEntity) {
         if (questionEntity instanceof MultipleChoiceQuestionEntity multipleChoiceQuestionEntity) {
-            return mapper.map(multipleChoiceQuestionEntity, MultipleChoiceQuestion.class);
+            return multipleChoiceQuestionEntityToDto(multipleChoiceQuestionEntity);
+        }
+        if (questionEntity instanceof AssociationQuestionEntity associationQuestionEntity) {
+            return associationQuestionEntityToDto(associationQuestionEntity);
```

Comment:
wouldn't it make more sense to use the modelMapper custom mapping functionality for all of this instead of this class?

Reply:
tbh idk  what the "custom mapping" functionality is

Reply:
would that make it significantly more simple?

Reply:
i mean you can leave it like this here but in the future i think it makes sense to use it. We already have a model mapper bean configuration class you'd just have to set the mapping functions there.

Reply:
> tbh idk what the "custom mapping" functionality is

It's what I used in the PR you reviewed earlier today.

On a modelMapper object you can do `mapper.typeMap(MyDTO.class, MyEntity.class).map(...)` to tell it how it should convert from the one type to the other

Reply:
I see. But we don't use that approach in other services either. 
If we use it properly, the whole mapping classes become obsolete, if I understand it correctly.
I.e., it requires are fundamental refactoring, first we need to think this through in my opinion.

Reply:
It's not necessary to change the existing stuff, because it does nothing except make the code cleaner. But it makes sense to use this from now on